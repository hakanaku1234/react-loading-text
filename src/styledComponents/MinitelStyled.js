import styled, { keyframes, css } from "styled-components";

export const WrapperMinitel = styled.div`
  position: relative;
  display: inline-block;
`;

const LineAnim = keyframes`
    from {
        width: 100%;
    }
    to {
        width: 0%;
    }
`;
const sizes = {
  md: 1200,
  sm: 992,
  xs: 768
};

const media = Object.keys(sizes).reduce((acc, label) => {
  acc[label] = (...args) => css`
    @media (max-width: ${sizes[label]}px) {
      ${css(...args)}
    }
  `;
  return acc;
}, {});

export const Line = styled.div`
    position: absolute;
    display:block;
    right:0;
    width:100%;
    z-index:999;
    transform-origin: 100% 0%;
    animation-name: ${LineAnim};
    animation-fill-mode: forwards;

    /* default */
    top: ${props => props.nb * props.style.default[0]}px;
    animation-delay: ${props =>
      props.style.default[2] + props.nb * props.style.default[3]}s;
    height: ${props => props.style.default[0]}px;
    background-color: ${props => props.style.default[1]};
    animation-duration: ${props => props.style.default[3]}s;
    animation-timing-function: ${props => props.style.default[4]};

    ${media.md` top: ${props => props.nb * props.style.md[0]}px;
                animation-delay: ${props =>
                  props.style.md[2] + props.nb * props.style.md[3]}s;
                height: ${props => props.style.md[0]}px;
                background-color: ${props => props.style.md[1]};
                animation-duration: ${props => props.style.md[3]}s;
                animation-timing-function: ${props => props.style.md[4]};
    `}
    ${media.sm` top: ${props => props.nb * props.style.sm[0]}px;
                animation-delay: ${props =>
                  props.style.sm[2] + props.nb * props.style.sm[3]}s;
                height: ${props => props.style.sm[0]}px;
                background-color: ${props => props.style.sm[1]};
                animation-duration: ${props => props.style.sm[3]}s;
                animation-timing-function: ${props => props.style.sm[4]};
    `}
    ${media.xs` top: ${props => props.nb * props.style.xs[0]}px;
                animation-delay: ${props =>
                  props.style.xs[2] + props.nb * props.style.xs[3]}s;
                height: ${props => props.style.xs[0]}px;
                background-color: ${props => props.style.xs[1]};
                animation-duration: ${props => props.style.xs[3]}s;
                animation-timing-function: ${props => props.style.xs[4]};
    `}  
`;
