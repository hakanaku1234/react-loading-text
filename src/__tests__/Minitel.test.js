import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import TestRenderer from "react-test-renderer";
import Minitel from "../index";
import { Line } from "../styledComponents/MinitelStyled";

Enzyme.configure({ adapter: new Adapter() });

jest.useFakeTimers();

describe("Minitel composant", () => {
  describe("When the composant is mount", () => {
    it("check if the callback is called", () => {
      const fnDefault = jest.fn();
      const style = {
        default: [18, 10, "white", 0, 0.1, "linear", fnDefault]
      };
      const Wrapper = shallow(
        <Minitel style={style}>
          <p>Hello World!</p>
        </Minitel>
      );

      expect(fnDefault).not.toBeCalled();
      jest.runAllTimers();
      expect(fnDefault).toHaveBeenCalledTimes(1);
    });

    it("check if the Line components are unmounted after the effect", () => {
      const fnDefault = jest.fn();
      const style = {
        default: [18, 10, "white", 0, 0.1, "linear", fnDefault]
      };
      const Wrapper = shallow(
        <Minitel style={style}>
          <p>Hello World!</p>
        </Minitel>
      );
      expect(Wrapper.children().find(Line)).toHaveLength(10);
      jest.runAllTimers();
      Wrapper.update();
      expect(Wrapper.children().find(Line)).toHaveLength(0);
    });

    it("renders correctly", () => {
      const fnDefault = jest.fn();
      const style = {
        default: [18, 10, "white", 0, 0.1, "linear", fnDefault]
      };
      const renderer = TestRenderer.create(
        <Minitel style={style}>
          <p>Hello World!</p>
        </Minitel>
      );
      expect(renderer.toJSON()).toMatchSnapshot();
    });
  });
});
