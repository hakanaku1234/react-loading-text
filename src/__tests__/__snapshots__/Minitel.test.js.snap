// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minitel composant When the composant is mount renders correctly 1`] = `
<div
  className="sc-bdVaJa jCeSos"
  style={
    Object {
      "default": Array [
        18,
        10,
        "white",
        0,
        0.1,
        "linear",
        [MockFunction],
      ],
    }
  }
>
  <div
    className="sc-bwzfXH eQSDlG"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH fGIjxC"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH ckPZck"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH bAlUyc"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH jrcLNc"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH fVsltN"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH cWyuic"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH fGigbh"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH dZvnEY"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <div
    className="sc-bwzfXH iGLMDd"
    style={
      Object {
        "default": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "md": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "sm": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
        "xs": Array [
          18,
          "white",
          0,
          0.1,
          "linear",
        ],
      }
    }
  />
  <p>
    Hello World!
  </p>
</div>
`;
